File: week17/week17-1.py
class Solution:
    def vowelStrings(self, words: List[str], queries: List[List[int]]) -> List[int]:
        vowel="aeiou"
        prefix=[]
        ans=[]
        n=len(words)
        now=0
        for word in words:
            if word[0] in vowel and word[-1] in vowel:now+=1
            prefix.append(now)
        for i,j in queries:
            if i==0:
                ans.append(prefix[j])
                continue
            ans.append(prefix[j]-prefix[i-1])
        return ans

File: week17/week17-1.cpp
class Solution {
public:
    vector<int> vowelStrings(vector<string>& words, vector<vector<int>>& queries) {
        string vowel="aeiou";
        vector<int> prefix;
        int now=0;
        for(auto word:words){
            int yes=0;
            for(auto v:vowel){
                if(word[0]==v)yes++;
                if(word.back()==v)yes++;
                if(yes==2)break;
            }
            if(yes==2)now++;
            prefix.push_back(now);
        }
        vector<int> ans;
        for(int k=0;k<queries.size();k++){
            int i=queries[k][0];
            int j=queries[k][1];
            if(i==0){
                ans.push_back(prefix[j]);
                continue;
            }
            ans.push_back(prefix[j]-prefix[i-1]);
        }
        return ans;
    }
};